### DISCUSSION TITLE
How to do pagination in codeCommit.listRepositories() ?

### DISCUSSION THREAD

**Discussion by rohitcoder** [aws/aws-sdk-js#4179] [Q&A]
I went through multiple blog posts and I also tried to review the source code of this SDK, it seems I can't do pagination here. Let's say i want to load only 10 repos and after that another 10 repos. How this would be possible?

I'm using below code snippet. i don't see any option to provide limit / offset value. Can someone guide me here?

```javascript
const repos = await codeCommit.listRepositories({
  sortBy: 'repositoryName',
  order: 'ascending',
}).promise()
```

BTW, I'm using from NPM JS

**Comment by ajredniwja** [comment]
Hey @rohitcoder thanks for opening this issue, looking at the [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CodeCommit.html#listRepositories-property ) unfortunately the `listRepositories` API does not provide an option to limit the results. 
Batch sizes are 1,000 for list repository operations. When the client sends the token back to AWS CodeCommit, another page of 1,000 records is retrieved.

The workaround around it would still need to call the API and just pick ten from that which I dont think you want to do.
